### DISCUSSION TITLE
Deleting problems, discussions, homeworks

### DISCUSSION THREAD

**Discussion by junukwon7** [hydro-dev/Hydro#41] [Q&A] [answered]
Is it possible?
https://github.com/vijos/vj4/pull/561/commits/459555f1a512863669a71fa1bf60a6b5f34b6428

**Comment by undefined-moe** [comment, marked as answer]
discussion deletion already supported.  
problem deletion might in next minor version.
homework deletion still needs some discussion. (also delete submissions for this homework? delete all progress and related discussions?)

**Reply by junukwon7** [reply]
How about hiding them and making new privileges for seeing them?
Maybe much easier and safer?

**Reply by undefined-moe** [reply]
but how to handle some stats? e.g. numAccept/numSubmit/rating

**Reply by junukwon7** [reply]
> but how to handle some stats? e.g. numAccept/numSubmit/rating

Example
submission itself: the code is hidden if the problem is hidden. 
rating, numAccept and numSubmit: counted as normal.

This is how my school's current OJ does.
Because the purpose of hiding was to hide problems/homeworks which were for testing.
We all don't want things like this.
![image](https://user-images.githubusercontent.com/48399106/104103064-80fc3c00-52e3-11eb-8917-734f1298bcd7.png)

**Reply by undefined-moe** [reply]
You mean create a `hidden homework` similar to the existing `hidden problem` function?

**Reply by junukwon7** [reply]
Yes. I just want them to be out of sights.
+ does the existing 'hidden problem' function fulfills the things I said? (hidden submission)

**Reply by undefined-moe** [reply]
it displays like this. (no detail access)
![image](https://user-images.githubusercontent.com/29992205/104103392-d41ec080-52dc-11eb-9961-5c8ac8835b66.png)

**Reply by undefined-moe** [reply]
I don't understand why homework need to be tested.  
A problem might be hidden for the following reason:

- Included in a contest so we need to ban discussion and solution for the specific problem
- Problem testdata needs to be tested
- Additional file or testdata not uploaded at current

However homework has a really simple logic, and all config are set before creating a homework.

**Reply by junukwon7** [reply]
Hmm. OK
Thanks.
